Document index.dimensions-based routing (#3229)
elastic/docs-content: manage-data/data-store/data-streams/time-series-data-stream-tsds.md

### Dimension-based routing [dimension-based-routing]

Within each TSDS backing index, {{es}} uses one of two strategies to route documents with the same dimensions to the same shards.

> 1. Based on the internally managed `index.dimensions` index setting (available as of stack version 9.2)

Reviewer: Note that this is the preferred one?

2. Based on the [`index.routing_path`](elasticsearch://reference/elasticsearch/index-settings/time-series.md#index-routing-path) index setting

The `index.dimensions`-based strategy uses a list of dimension paths that's automatically kept up-to-date and is not user-configurable.

> When active, the routing decision will be made based on all dimension fields instead of only those defined statically via the `index.routing_path` setting,

Reviewer: ```suggestion
When active, the routing decision will be made based on all dimension fields, instead of only those defined statically via the `index.routing_path` setting,
```

> which can help to prevent shard hot-spotting issues.

Reviewer: I'd leave the latter out.. doesn't add much, we don't need to go into details why this is better.

> This strategy is only available on data streams and requires that there are no dynamic templates that set `time_series_dimension: true`.

Reviewer: ```suggestion
This strategy is only available on data streams with no dynamic templates setting `time_series_dimension: true`.
```
